src/main/java/org/example/services/UserInfoInputService.java

```diff
@@ -0,0 +1,33 @@
+package org.example.services;
+
+import java.util.Scanner;
+
+public class UserInfoInputService {
+  Scanner scanner = new Scanner(System.in);
+
+  public void userInfoInputPrompt(AddUserDetailsService addUserDetailsService) {
```

Comment:
AddUserDetailsService already has dependency on UserInfoInputService and you're passing AddUserDetailsService again here. Can we make userInfoInputPrompt take input and create a object and return back to addUserDetailsService

Reply:
Removed dependency of UserInfoInputService on AddUserDetailsService , now function of UserInfoInputService ( userInfoInputPrompt ) is returning an object to AddUserDetailsService